.gitignore

```diff
@@ -1,5 +1,29 @@
-# Ensure only custom code is contributed.
+##
+## Temporary files generated by various OSs and IDEs
```

Comment:
These should be put in a global git ignore rather than having to manage it for every project.

https://help.github.com/articles/ignoring-files/#create-a-global-gitignore